hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -689,4 +694,23 @@ public long countActionsAll() {
         return distributionSetRepository.findInstalledAtTarget(controllerId);
     }
 
+    @Override
+    public int deleteActionsByStatusAndLastModifiedBefore(final Set<Status> status, final long lastModified) {
+        if (status.isEmpty()) {
+            return 0;
+        }
+        /*
+         * We use a native query here because Spring JPA does not support to
+         * specify a LIMIT clause on a DELETE statement. However, for this
+         * specific use case (action cleanup), we must specify a row limit to
+         * reduce the overall load on the database.
+         */
+        final String queryStr = String.format(QUERY_DELETE_ACTIONS_BY_STATE_AND_LAST_MODIFIED,
```

Comment:
I would prefer the use of a prepared statement over a handcrafted string. Maybe like `JpaControllerManagement#setLastTargetQuery(...)`

Reply:
Done